### ISSUE TITLE
Question: md-select infinite scroll

### ISSUE THREAD

**Issue by Xandroid4Net** [angular/material#9915]
Is there a way to compbine md-select with md-virtual-repeat? I have not been able to find an example of how to create a md-select with the ability to lazy load items on scroll.

**Comment by djleonskennedy** [comment]
agree, any work around to implementation of this?

NOTE: have idea to use panel to implement it

**Comment by Xandroid4Net** [comment]
Here is a demo page that has a lazy load md select. Not exactly what I was looking for but it may help others.

<html lang="en">

<head>
	<meta name="viewport" content="width=device-width, initial-scale=1">
	
	<link rel="stylesheet" href="http://ajax.googleapis.com/ajax/libs/angular_material/1.1.0/angular-material.min.css">
</head>

<body ng-app="app" ng-cloak>
	<div ng-controller="main">
		<md-input-container class="md-block" flex-gt-sm>
			<label>State</label>
			<md-select ng-model="user.state" whenscrollends="loadMore()">
				<md-option ng-repeat="state in states" value="{{state}}">
					{{state}}
				</md-option>
			</md-select>
		</md-input-container>

    

	</div>
	
	<script src="http://ajax.googleapis.com/ajax/libs/angularjs/1.5.5/angular.min.js"></script>
	<script src="http://ajax.googleapis.com/ajax/libs/angularjs/1.5.5/angular-animate.min.js"></script>
	<script src="http://ajax.googleapis.com/ajax/libs/angularjs/1.5.5/angular-aria.min.js"></script>
	<script src="http://ajax.googleapis.com/ajax/libs/angularjs/1.5.5/angular-messages.min.js"></script>

	
	<script src="http://ajax.googleapis.com/ajax/libs/angular_material/1.1.0/angular-material.min.js"></script>

	
	<script type="text/javascript">    
    /**
     * You must include the dependency on 'ngMaterial' 
     */
    angular.module('app', ['ngMaterial']).controller('main', function($scope) {
      $scope.states=['A','B','C','D','E','F','G','H','I','J','K'];
      var i =0;
      $scope.loadMore = function()
      {
        var a = [];
          for(var j=0;j<10;j++)
          {
            a.push(i);
            i+=1;
          }
          $scope.states.push.apply($scope.states,a);       
      }
    }).directive("whenscrollends", function() {
        return {
            restrict: "A",
            link: function(scope, element, attrs) {
                var container = element.find('md-content');
                container.css('height','300px');
                var visibleHeight = 300;
                var threshold = 100;

                container.on('scroll',function(e){
                  var scrollableHeight = container.prop('scrollHeight');
                    var hiddenContentHeight = scrollableHeight - visibleHeight;
                    if (hiddenContentHeight - container[0].scrollTop <= threshold) {
                            scope.$apply(attrs.whenscrollends);
                    }
                });
               
            }
        }
    });
  </script>

</body>

</html>

**Comment by Xandroid4Net** [comment]
This one is based on the autocomplete and is a little bit closer but it is not perfect.

http://codepen.io/xandroid4net/pen/BQpKOo

**Comment by riyask** [comment]
I tried this https://codepen.io/xandroid4net/pen/jVyqzV
It is working fine but when we implement it using serer side API, When it reaches to scroll end, it calls the API four or five times. Any one knows the solution

**Comment by Splaktar** [comment]
Thank you for taking the time to submit this issue. However there are better avenues for troubleshooting or asking for help with your app. These include [Stack Overflow](https://stackoverflow.com/), [Gitter](https://gitter.im/), this project's [mailing list or forums](https://groups.google.com/forum/#!forum/ngmaterial), etc.

Please limit GitHub issues to bug reports and feature requests.
